redpanda/templates/configmap.yaml

```diff
@@ -26,25 +26,109 @@ metadata:
     app.kubernetes.io/instance: {{ .Release.Name | quote }}
     app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
     app.kubernetes.io/component: {{ template "redpanda.name" . }}
-  {{- with .Values.labels }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+{{- with .Values.commonLabels }}
+  {{- toYaml . | nindent 4 }}
+{{- end }}
 data:
   redpanda.yaml: |
     config_file: /etc/redpanda/redpanda.yaml
-    license_key: {{ .Values.config.license_key }}
-    organization: {{ .Values.config.organization }}
+    {{- if gt (len .Values.license_key) 0 }}
+    license_key: {{ .Values.license_key }}
+    {{- end }}
+    {{- if gt (len .Values.organization) 0 }}
+    organization: {{ .Values.organization }}
+    {{- end }}
     redpanda:
-{{ toYaml .Values.config.redpanda | indent 6 }}
+{{- if .Values.auth.sasl.enabled }}
+  {{- with .Values.auth.sasl.users }}
+      superusers:
+    {{ toYaml . | indent 8 }}
+  {{- end }}
+{{- end }}
+{{- if .Values.listeners.admin.enabled }}
+      admin:
+        address: {{ .Values.listeners.admin.address }}
+        name: admin
+        port: {{ .Values.listeners.admin.port }}
```

Comment:
admin is a list: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/config/schema.go#L55  we should append the `-`

```
      admin:
        - address: {{ .Values.listeners.admin.address }}
          name: admin
          port: {{ .Values.listeners.admin.port }}
```

Reply:
This means there is a bug in the documentation: https://docs.redpanda.com/docs/21.11/cluster-administration/configuration/

This applies to the following `redpanda.yaml` sections:
- admin
- admin_api_tls
- rpc_server
- rpc_server_tls

I will update `values.yaml` and all impacted templates, test, and then include that update in this branch.

Reply:
Both forms are supported: https://github.com/redpanda-data/redpanda/blob/084862a45cd92449d877622dc06e42f77df31ceb/src/v/config/node_config.h#L38 - list is probably preferred.

Reply:
> Both forms are supported: https://github.com/redpanda-data/redpanda/blob/084862a45cd92449d877622dc06e42f77df31ceb/src/v/config/node_config.h#L38 - list is probably preferred.

This is good news (thanks Ben). In that case I would like to still make this change (treating admin and rpc listeners as a group rather than only a single listener), but I would rather do this in a separate PR. I created a ticket to track this: https://github.com/redpanda-data/helm-charts/issues/111

Reply:
For some context, although list vs. item is supported in redpanda, we plan to slowly deprecate single-item support in rpk.